jre/java/java/util/stream/Collectors.java

```diff
@@ -170,6 +170,38 @@ public static Collector<CharSequence,?,String> joining(CharSequence delimiter) {
         downstream.finisher());
   }
 
+  public static <T, U, A, R> Collector<T, ?, R> flatMapping(Function<? super T,? extends Stream<?
+      extends U>> mapper, Collector<? super U, A, R> downstream) {
+    return new CollectorImpl<>(
+        downstream.supplier(),
+        (A a, T t) -> {
```

Comment:
Per JavaDoc:
Each mapped stream is closed after its contents have been placed downstream.

Let's add a test for it.